execute/report/report.go

```diff
@@ -28,12 +28,16 @@ func buildSingleChainReport(
 	lggr logger.Logger,
 	hasher cciptypes.MessageHasher,
 	tokenDataReader types.TokenDataReader,
-	encoder cciptypes.ExecutePluginCodec,
 	report plugintypes.ExecutePluginCommitDataWithMessages,
-	maxMessages int,
-) (cciptypes.ExecutePluginReportSingleChain, int, error) {
-	if maxMessages == 0 {
-		maxMessages = len(report.Messages)
+	messages map[int]struct{},
```

Comment:
Switched from taking the first N messages, to taking specific messages. The "executed" logic will be moved into `checkMessage`.